### PR TITLE
Classify all models in pipeline

### PR DESCRIPTION
**Pull request by gferraro**
Change to let pipeline classify all models, moved model config into pipeline config file
reliant on https://github.com/TheCacophonyProject/classifier-pipeline/pull/139

### CODE REVIEW

File: processing/thermal.py [hunk 1]
```diff
@@ -60,30 +60,18 @@ def process(recording, conf):
...
+    command = "{}".format(command)
     classify_info = run_classify_command(command, conf.classify_dir)
 
-    track_info = classify_info["tracks"]
-    formatted_tracks = format_track_data(track_info)
+    format_track_data(classify_info["tracks"])
 
     # Auto tag the video
-    tagged_tracks, tags = calculate_tags(formatted_tracks, conf)
+    algorithm_id = api.get_algorithm_id(classify_info["algorithm"])
```
Comment (clare) [inline processing/thermal.py:70]: The algorithm id should load information specific to each algorithm.   So if there are multiple models that are being processed I thought you would need mulitple algorithm ids for each one.   I guess they could all use the same one, (as long as it is obvious from the names what applies to what). but it's not how it was intended.

File: processing/thermal.py [hunk 2]
```diff
@@ -185,124 +172,122 @@ def update_metadata(conf, recording, api):
...
+        )
+        model_results = []
+        for model_result in track["predictions"]:
+            added, tag = add_track_tag(api, recording, track, model_result, logger)
             if added:
-                model_tags.append((model, track_to_save))
+                model_results.append(
+                    (classify_result.models_by_id[model_result["id"]], model_result)
+                )
+        model_result = get_master_tag(model_results, wallaby_device)
```
Comment (clare) [inline processing/thermal.py:187]: Reusing model_result variable name here is a bit confusing and could cause bugs.   Probably master_result or somethimg similar would be a better name.

**Review by clare — CHANGES_REQUESTED** [review]
Just a couple of things to make sure you tidy up.    (Like the duplicate code).

File: processing/tagger.py [hunk 3]
```diff
@@ -16,18 +17,16 @@
...
-    if len(tags) == 0:
-        tags[FALSE_POSITIVE] = FALSE_POSITIVE_TAG
-    else:
-        multiple_confidence = calculate_multiple_animal_confidence(
-            clear_animals + unclear_animals
-        )
-        if multiple_confidence > conf.min_confidence:
-            tags[MULTIPLE] = {"event": MULTIPLE, CONFIDENCE: multiple_confidence}
+    multiple_confidence = calculate_multiple_animal_confidence(
+        clear_animals + unclear_animals
```
[the remark this answers is not in the document]
Comment (gferraro) [reply processing/tagger.py:26]: We weren't using it ever so i removed it

File: processing/tagger.py [hunk 4]
```diff
@@ -74,31 +83,35 @@ def get_significant_tracks(tracks, conf):
...
-                    )
                 else:
-                    tags[tag] = {CONFIDENCE: track[CONFIDENCE]}
-
-            else:
-                unclear_animals.append(track)
-                tags[UNIDENTIFIED] = {CONFIDENCE: DEFAULT_CONFIDENCE}
-                track[TAG] = UNIDENTIFIED
+                    tags[UNIDENTIFIED] = {CONFIDENCE: DEFAULT_CONFIDENCE}
+                    unclear_animals.append(track)
```
[the remark this answers is not in the document]
Comment (gferraro) [reply processing/tagger.py:113]: should be track but have changed variables names to represent this e.g. `unclear_tracks` and `clear_tracks`

File: processing/thermal.py [hunk 5]
```diff
@@ -60,30 +60,18 @@ def process(recording, conf):
...
-        model_results.append(model_result)
-
-    return model_results
-
-
-def classify_file(api, command, conf, model):
-
-    command = "{} -m {} -p {}".format(command, model.model_file, model.preview)
+def classify_file(api, command, conf):
+    command = "{}".format(command)
```
[the remark this answers is not in the document]
Comment (gferraro) [reply processing/thermal.py:64]: :)

File: processing/thermal.py (hunk 2 above)
Comment (gferraro) [reply processing/thermal.py:187]: good point

File: processing/thermal.py (hunk 1 above)
Comment (gferraro) [reply processing/thermal.py:70]: I think there is no need as the only thing different in each is model name and model file which is being saved in the data, so algorithm_id i am using to represent the tracking algorithm. I think this makes more sense

Comment (gferraro) [reply processing/thermal.py:70]: Especially since it applies to the tracks

**Review by clare — APPROVED** [review]
Thanks GP.   Sorry for the delay

**Pull request merged**